frontends/web/src/utils/request.ts

```diff
@@ -35,18 +36,31 @@ function extConfig(key, defaultValue) {
 export const apiPort = extConfig('{{ API_PORT }}', '8082');
 export const apiToken = extConfig('{{ API_TOKEN }}', '');
 
-export function isTLS() {
+export function isTLS(): boolean {
   return document.URL.startsWith('https://');
 }
 
-export function apiURL(endpoint) {
+export function apiURL(endpoint: string): string {
   return (isTLS() ? 'https://' : 'http://') + 'localhost:' + apiPort + '/api/' + endpoint;
 }
 
-function handleError(endpoint) {
-  return function(json) {
+// https://fettblog.eu/typescript-hasownproperty/
```

Comment:
pls add a bit more docstring about the purpose of this, in case this page becomes unavailable.